cla_backend/apps/cla_eventlog/tests/test_find_and_delete_old_cases.py

```diff
@@ -0,0 +1,59 @@
+from django.test import TestCase
+from django.utils import timezone
+from dateutil.relativedelta import relativedelta
+from freezegun import freeze_time
+
+from core.tests.mommy_utils import make_recipe
+from cla_eventlog.models import Log
+from legalaid.models import Case
+
+
+class FindAndDeleteOldCases(TestCase):
+    def create_three_year_old_case(self):
+        case = None
+        freezer = freeze_time(timezone.now() + relativedelta(years=-3))
+        freezer.start()
+        case = make_recipe("legalaid.case")
+        freezer.stop()
+        return case
+
+    def create_event_log_for_case(self, case, code, created=None):
+        if created:
+            make_recipe("cla_eventlog.log", case=case, code=code, created=created)
+        else:
+            make_recipe("cla_eventlog.log", case=case, code=code)
+
+    def test_find_cases_viewed_two_years_ago(self):
+        new_case = make_recipe("legalaid.case")
+        old_case_1 = self.create_three_year_old_case()
+        old_case_2 = self.create_three_year_old_case()
+
+        self.assertEqual(len(Case.objects.all()), 3)
+
+        self.create_event_log_for_case(new_case, "CASE_VIEWED")
+        self.create_event_log_for_case(new_case, "CASE_CREATED")
+        self.create_event_log_for_case(new_case, "CASE_VIEWED", created=timezone.now() + relativedelta(years=-3))
+
+        self.create_event_log_for_case(old_case_1, "CASE_VIEWED")
+        self.create_event_log_for_case(old_case_1, "MT_CHANGED")
+        self.create_event_log_for_case(old_case_1, "CASE_VIEWED", created=timezone.now() + relativedelta(years=-1))
+
+        self.create_event_log_for_case(old_case_2, "CASE_VIEWED", created=timezone.now() + relativedelta(years=-3))
+        self.create_event_log_for_case(old_case_2, "CASE_CREATED", created=timezone.now() + relativedelta(years=-3))
+        self.create_event_log_for_case(old_case_2, "CASE_VIEWED", created=timezone.now() + relativedelta(years=-2))
+
+        logs = Log.objects.all()
+
+        self.assertEqual(len(logs), 9)
+
+        event_logs_created_two_years_ago = Log.objects.filter(
+            # Cases created two years ago or over
+            case__created__lte=timezone.now() + relativedelta(years=-2),
+            # That have logs created two years ago or over
+            created__lte=timezone.now() + relativedelta(years=-2),
```

Comment:
This tells us about cases that have logs older than two years, but nothing about whether they have logs newer than two years ago, right?
ie if a case was viewed three years ago and also last week, you would return the three year old log

Reply:
Yeah good point, I'll change this now